### PR TITLE
Move test_assets and logits generation script inside tests directory

### PR DESCRIPTION
**Pull request by SurbhiJainUSC**
# Description

This PR refactors tests assets based on https://screenshot.googleplex.com/574uqLTPrQy8FuN

* Move `src/MaxText/test_assets` to `tests/assets/golden_logits`
* Move golden data generation scripts inside `src/MaxText/scratch_code/` to `tests/assets/logits_generation`
* Updated Github CI workflows to reflect this new tests assets path
* Updated Dockerfiles to reflect this new tests assets path

# Tests
Unit/integ tests

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by hengtaoguo — APPROVED** [review]
Could you also update these two lines regarding the test_image.jpg/test_video.mp4 path? This is the most critical one at the moment. Thanks!

https://github.com/AI-Hypercomputer/maxtext/blob/111cb6b3f3d9037cd74d0e41504483092e22cb1e/tests/check_qwen3_embedding_vs_reference.py#L534-L535

There might be some other places referencing these static assets too:

https://github.com/search?q=repo%3AAI-Hypercomputer%2Fmaxtext+test_image.jpg&type=code

**Comment by SurbhiJainUSC** [comment]
> Could you also update these two lines regarding the test_image.jpg/test_video.mp4 path? This is the most critical one at the moment. Thanks!
> 
> https://github.com/AI-Hypercomputer/maxtext/blob/111cb6b3f3d9037cd74d0e41504483092e22cb1e/tests/check_qwen3_embedding_vs_reference.py#L534-L535
> 
> There might be some other places referencing these static assets too:
> 
> https://github.com/search?q=repo%3AAI-Hypercomputer%2Fmaxtext+test_image.jpg&type=code

Thank you Hengtao, updated.

File: end_to_end/tpu/deepseek/v3-671b/2_test_deepseek.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@
 # 2. Run logit check, pre-training, fine-tuning, and decoding.
 
 # The golden logit can be generated by:
-# python3 -m MaxText.scratch_code.generate_hf_golden_logits --model-id=deepseek-ai/DeepSeek-V3 --output-path=golden_data_deepseek3-671b.jsonl --prompts='I love to' --hf-model-path=$local_bf16_path --trust-remote-code=False --hf-load-dtype=bfloat16 
+# python3 tests/assets/logits_generation/generate_hf_golden_logits.py --model-id=deepseek-ai/DeepSeek-V3 --output-path=golden_data_deepseek3-671b.jsonl --prompts='I love to' --hf-model-path=$local_bf16_path --trust-remote-code=False --hf-load-dtype=bfloat16
```
Comment (shralex) [inline end_to_end/tpu/deepseek/v3-671b/2_test_deepseek.sh]: why not module syntax ? python3 -m with "." instead of directory path

File: end_to_end/tpu/gpt_oss/120b/test_gpt_oss.sh [hunk 2]
```diff
@@ -10,7 +10,7 @@
 # Example Usage: export HF_TOKEN=<huggingface_access_token>; export BASE_OUTPUT_PATH=<GCS_bucket_path>; bash test_gpt_oss.sh
 
 # The golden logit can be generated by:
-# python3 -m MaxText.scratch_code.generate_hf_golden_logits --model-id=openai/gpt-oss-120b --output-path=golden_data_gpt-oss-120b.jsonl --prompts='I love to;Today is a;What is the' --hf-model-path=<local_bf16_path>
+# python3 tests/assets/logits_generation/generate_hf_golden_logits.py --model-id=openai/gpt-oss-120b --output-path=golden_data_gpt-oss-120b.jsonl --prompts='I love to;Today is a;What is the' --hf-model-path=<local_bf16_path>
```
Comment (shralex) [inline end_to_end/tpu/gpt_oss/120b/test_gpt_oss.sh]: same comment

File: end_to_end/tpu/deepseek/v3-671b/2_test_deepseek.sh (hunk 1 above)
Comment (SurbhiJainUSC) [reply end_to_end/tpu/deepseek/v3-671b/2_test_deepseek.sh]: Updated

File: end_to_end/tpu/gpt_oss/120b/test_gpt_oss.sh (hunk 2 above)
Comment (SurbhiJainUSC) [reply end_to_end/tpu/gpt_oss/120b/test_gpt_oss.sh]: Done

**Review by dipannita08 — APPROVED** [review]

File: dependencies/dockerfiles/maxtext_runner.Dockerfile [hunk 3]
```diff
@@ -6,7 +6,7 @@ FROM $BASEIMAGE
 #FROM maxtext_base_image
 
 ENV MAXTEXT_ASSETS_ROOT=/deps/src/MaxText/assets
-ENV MAXTEXT_TEST_ASSETS_ROOT=/deps/src/MaxText/test_assets
+ENV MAXTEXT_TEST_ASSETS_ROOT=/deps/tests/assets
```
Comment (shralex) [inline dependencies/dockerfiles/maxtext_runner.Dockerfile:9]: IIUC this changes the location of the files in the docker image. Are there places in ml-automation that need to be updated ?

File: end_to_end/tpu/deepseek/v2-16b/test_deepseek.sh [hunk 4]
```diff
@@ -10,7 +10,7 @@
 # Example Usage: export HF_TOKEN=<huggingface_access_token>; export BASE_OUTPUT_PATH=<GCS_bucket_path>; bash test_deepseek.sh
 
 # The golden logit can be generated by:
-# python3 -m MaxText.scratch_code.generate_hf_golden_logits --model-id=deepseek-ai/DeepSeek-V2-Lite --output-path=golden_data_deepseek2-16b.jsonl --prompts='I love to;Today is a;What is the' --hf-model-path=$local_bf16_path --trust-remote-code=False
+# python3 -m tests.assets.logits_generation.generate_hf_golden_logits --model-id=deepseek-ai/DeepSeek-V2-Lite --output-path=golden_data_deepseek2-16b.jsonl --prompts='I love to;Today is a;What is the' --hf-model-path=$local_bf16_path --trust-remote-code=False
```
Comment (shralex) [inline end_to_end/tpu/deepseek/v2-16b/test_deepseek.sh:13]: lets make a note of this change for our release notes, since this might be breaking

File: dependencies/dockerfiles/maxtext_runner.Dockerfile (hunk 3 above)
Comment (SurbhiJainUSC) [reply dependencies/dockerfiles/maxtext_runner.Dockerfile:9]: No, these test assets are accessed by end-to-end scripts in `end_to_end/` in MaxText and I have already updated these scripts in this PR. DAGs in `ml-auto-solutions` repo calls this end-to-end scripts.
